### PR TITLE
Extensions: Fix settings link visibility and redirection issues

### PR DESCRIPTION
**Pull request by ice9js**
This PR aims to improve the functionality of extension settings link by implementing the following:

- Don't show the link before a plugin is installed on the currently selected site.
- Show the link when the plugin is installed, regardless if it's active or not.
- In case no site is selected, make it redirect to site selection.

screen shot 2017-07-13 at 21 00 47

# Testing

- Head over to the details page of a plugin with a calypso extension ( WPSC, WPJM, Zoninator ) on a free site and make sure the link isn't visible.
- Switch to a Jetpack/Business site without the plugin installed and make sure the link isn't visible.
- Switch to a Jetpack/Business site with the plugin installed and make sure the link is visible and working.
- Try (de)activating the plugin and verify the link remains visible.
- Open the plugin details page without any site selected and verify the link is visible and redirects to site selection.

### CODE REVIEW

File: client/my-sites/plugins/plugin-meta/index.jsx [hunk 1]
```diff
@@ -459,7 +459,7 @@ class PluginMeta extends Component {
 
 				{ path &&
 					<CompactCard
-						href={ `${ path }/${ this.props.slug }` }>
+						href={ trimEnd( `${ path }/${ this.props.slug || '' }`, '/' ) }>
```
Comment (ockham) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Can we use [`addSiteFragment`](https://github.com/Automattic/wp-calypso/blob/a70587157b33783ba4d5e0e4a1e136680d459d51/client/lib/route/path.js#L50-L64) instead (and fix it, if needed)?

**Comment by ockham** [comment]
Some confusion about the PR desc:

> * Don't show the link if the site it's a Business/Jetpack site.

Shouldn't the criterion be "Show the link if the site _is_ a Jetpack site"? How does the site plan (Business) enter into this?

> * Show the link when the plugin is installed, regardless if it's active or not.

Shouldn't we hide the link when the plugin is inactive? Cf. `wp-admin`.

**Comment by ice9js** [comment]
>Shouldn't the criterion be "Show the link if the site is a Jetpack site"? How does the site plan (Business) enter into this?

I might have made a typo. But now that I think of it, that criterion is kind of useless as it should only show when the plugin is installed for which you need Jetpack anyway.

>Shouldn't we hide the link when the plugin is inactive? Cf. wp-admin.

That depends. I was thinking that in some cases a user might want to check some old settings of a deactivated plugin. But for now, I think we can narrow it down to just the active plugins.

File: client/my-sites/plugins/plugin-meta/index.jsx [hunk 2]
```diff
@@ -436,7 +436,7 @@ class PluginMeta extends Component {
 		} );
 
 		const plugin = this.props.selectedSite && this.props.sites[ 0 ] ? this.props.sites[ 0 ].plugin : this.props.plugin;
-		const path = this.getExtensionSettingsPath( plugin );
+		const path = ( ! this.props.selectedSite || this.props.isInstalledOnSite ) && this.getExtensionSettingsPath( plugin );
```
Comment (ockham) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Why `! this.props.selectedSite`? This means hiding the path when a site is selected, doesn't it?

**Comment by ockham** [comment]
>> Shouldn't we hide the link when the plugin is inactive? Cf. wp-admin.

> That depends. I was thinking that in some cases a user might want to check some old settings of a deactivated plugin. But for now, I think we can narrow it down to just the active plugins.

Yeah, I'd strongly opt for parity with `wp-admin` 😉

File: client/my-sites/plugins/plugin-meta/index.jsx (hunk 2 above)
Comment (ice9js) [inline client/my-sites/plugins/plugin-meta/index.jsx]: The whole thing only hides the path when a site is selected *and* the plugin is not installed.

Comment (ockham) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Okay, right. But let's still change the criterion to check if a plugin is _active_.

**Comment by donnapep** [comment]
@ice9js Let's clarify some terminology to reduce confusion between "installed" vs. "activated". If a plugin is not installed then you won't see it in the plugins list at all. So we don't need to do anything in this case because you won't even be able to get to the plugins detail page if it's not installed.

Given that, this is the criteria that should be tested in my opinion -

Show the link when:
- A Calypso extension has a `settings_path` and is active.

Do not show the link when:
- A plugin is not a Calypso extension (e.g. Akismet).
- A Calypso extension is active but does not have a `settings_path` set in `package.json` (e.g. WooCommerce).
- A Calypso extension has a `settings_path` set in `package.json` but is not active (this is the bit that this PR seeks to address I believe).

**Comment by donnapep** [comment]
> In case no site is selected, make it redirect to site selection.

You shouldn't need to do anything special here. It's the responsibility of the plugin to show the site selector.

**Comment by donnapep** [comment]
> Switch to a Jetpack/Business site without the plugin installed and make sure the link isn't visible.

Ah, right. So you can still get to that page by switching sites. OK, so we do still need the installed check + the active check.

**Comment by ice9js** [comment]
>> In case no site is selected, make it redirect to site selection.

>You shouldn't need to do anything special here. It's the responsibility of the plugin to show the site selector.

That's what I meant and I'm sorry for poor wording :( But I did need to touch that redirection as it was previously sending me to `/extensions/{extension-name-here}/null`.

File: client/my-sites/plugins/plugin-meta/index.jsx (hunk 2 above)
Comment (ice9js) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Addressed in [0306974](https://github.com/Automattic/wp-calypso/pull/16222/commits/0306974b364f4b9df5ee6a9e5861e6a35971178b)

**Comment by ice9js** [comment]
I ensured the link is only shown for active plugins instead of all that are installed.
With that, this should be ready to go unless there are any other objections 🙂

**Pull request merged**